### PR TITLE
[New Hack] 071-GitHub Copilot

### PR DESCRIPTION
**Pull request by Whowong**
Submitting new WTH for GitHub Copilot

### CODE REVIEW

File: 071-GitHubCopilot/Coach/Solution-00.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Challenge 00 - <Title of Challenge> - Coach's Guide
```
Comment (jordanbean-msft) [inline 071-GitHubCopilot/Coach/Solution-00.md:1]: what should this title be?

File: 071-GitHubCopilot/Coach/Solution-00.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Challenge 00 - <Title of Challenge> - Coach's Guide 
+
+**[Home](./README.md)** - [Next Solution >](./Solution-01.md)
+
+## Notes & Guidance
+
+A GitHub account, GitHub Copilot, and VSCode is all that is needed for this hack.  All are free and GitHub Copilot if needed has a free 30 day trial.
```
Comment (jordanbean-msft) [inline 071-GitHubCopilot/Coach/Solution-00.md:7]: VS Code

File: 071-GitHubCopilot/Coach/Solution-03.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+## Notes & Guidance
+
+- This challenge is to show that Copilot can do more than just help with developers writing code.  Here we are going to also leverage Copilot to help with commit messages and a GitHub Action pipeline (yaml).  At the end of this challenge you can also discuss some other areas where Copilot may be able to help.  Copilot chat will likely be able to help scaffold a good portion of this.  Below are some examples:
+  - Helping with XML such as in APIM policies
+  - Infrastructure as code such as ARM/Bicep
+  - SQL Queries
+- To generate a commit message, they can go to VSCode, and in the source control section there will be these stars/diamonds next to the commit message box.  Selecting that will allow you to leverage Copilot to generate a commit message.  They can generate this after creating the pipeline to have something to checkin to.
+- This could also be a great time to talk about some of GitHub Copilot Enterprises other features such as PR message creation which is along the same concept. However this is a separate license and not part of the hack.
+[Sample Pipeline Solution ](./Solutions/Solution-03.yaml)
```
Comment (jordanbean-msft) [inline 071-GitHubCopilot/Coach/Solution-03.md:13]: remove extra space

File: 071-GitHubCopilot/Coach/Solution-03.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+[< Previous Solution](./Solution-02.md) - **[Home](./README.md)** - [Next Solution >](./Solution-04.md)
+
+## Notes & Guidance
+
+- This challenge is to show that Copilot can do more than just help with developers writing code.  Here we are going to also leverage Copilot to help with commit messages and a GitHub Action pipeline (yaml).  At the end of this challenge you can also discuss some other areas where Copilot may be able to help.  Copilot chat will likely be able to help scaffold a good portion of this.  Below are some examples:
+  - Helping with XML such as in APIM policies
+  - Infrastructure as code such as ARM/Bicep
+  - SQL Queries
+- To generate a commit message, they can go to VSCode, and in the source control section there will be these stars/diamonds next to the commit message box.  Selecting that will allow you to leverage Copilot to generate a commit message.  They can generate this after creating the pipeline to have something to checkin to.
+- This could also be a great time to talk about some of GitHub Copilot Enterprises other features such as PR message creation which is along the same concept. However this is a separate license and not part of the hack.
```
Comment (jordanbean-msft) [inline 071-GitHubCopilot/Coach/Solution-03.md:12]: should you be more specific about the license required?

File: 071-GitHubCopilot/Coach/Solution-05.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Challenge 05 - Debugging and Code Optimization with Copilot - Coach's Guide 
+
+[< Previous Solution](./Solution-04.md) - **[Home](./README.md)**
+
+## Notes & Guidance
+
+- Students can grab a file of their choice from the juice-shop repo.  For example you can choose the order.ts file located under routes folder.
```
Comment (jordanbean-msft) [inline 071-GitHubCopilot/Coach/Solution-05.md:7]: should you include links to this repo & file?

File: 071-GitHubCopilot/Student/Challenge-00.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+Thank you for participating in the GitHub Copilot What The Hack. Before you can hack, you will need to set up some prerequisites.
+
+## Common Prerequisites
+
+We have compiled a list of common tools and software that will come in handy to complete most What The Hack Azure-based hacks!
+
+You might not need all of them for the hack you are participating in. However, if you work with Azure on a regular basis, these are all things you should consider having in your toolbox.
+
+<!-- If you are editing this template manually, be aware that these links are only designed to work if this Markdown file is in the /xxx-HackName/Student/ folder of your hack. -->
```
Comment (jordanbean-msft) [inline 071-GitHubCopilot/Student/Challenge-00.md:15]: should you remove this comment?

File: 071-GitHubCopilot/Student/Challenge-05.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# Challenge 05 - Debugging and Code Optimization with Copilot
+
+[< Previous Challenge](./Challenge-04.md) - **[Home](../README.md)** 
+
+## Introduction
+In this final challenge, you'll harness the power of GitHub Copilot to identify and resolve security vulnerabilities in a pre-written code snippet*. You will also optimize your code, refining its performance and maintainability based on the intelligent suggestions provided by Copilot. This challenge is an opportunity to deepen your understanding of how Copilot can be an integral part of your debugging and code optimization processes.
```
Comment (jordanbean-msft) [inline 071-GitHubCopilot/Student/Challenge-05.md:6]: why the extra "*"?

File: 071-GitHubCopilot/Student/Challenge-05.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+*Please note, while GitHub Copilot can identify vulnerabilities and filter out security vulnerabilities in its suggestions [Link](https://github.blog/2023-02-14-github-copilot-now-has-a-better-ai-model-and-new-capabilities/), it does not replace traditional security tools such as GitHub Advanced Security which have more robust capabilities at identifying vulnerabilities.
```
Comment (jordanbean-msft) [inline 071-GitHubCopilot/Student/Challenge-05.md:8]: should this have a closing "*"?

File: 071-GitHubCopilot/Coach/Solution-00.md (hunk 1 above)
Comment (Whowong) [reply 071-GitHubCopilot/Coach/Solution-00.md]: Updated to include title: Prerequisites - Ready, Set, GO! - Coach's Guide

File: 071-GitHubCopilot/Coach/Solution-00.md (hunk 2 above)
Comment (Whowong) [reply 071-GitHubCopilot/Coach/Solution-00.md]: Fixed Typo

File: 071-GitHubCopilot/Coach/Solution-03.md (hunk 3 above)
Comment (Whowong) [reply 071-GitHubCopilot/Coach/Solution-03.md]: Done

File: 071-GitHubCopilot/Coach/Solution-03.md (hunk 4 above)
Comment (Whowong) [reply 071-GitHubCopilot/Coach/Solution-03.md]: Added additional context for licenses and the different types

File: 071-GitHubCopilot/Coach/Solution-05.md (hunk 5 above)
Comment (Whowong) [reply 071-GitHubCopilot/Coach/Solution-05.md]: Good idea, I have added them

File: 071-GitHubCopilot/Student/Challenge-00.md (hunk 6 above)
Comment (Whowong) [reply 071-GitHubCopilot/Student/Challenge-00.md]: Removed

File: 071-GitHubCopilot/Student/Challenge-05.md (hunk 7 above)
Comment (Whowong) [reply 071-GitHubCopilot/Student/Challenge-05.md:6]: My original intent was to make sure students were aware that this was not a security tool replacement.  However looking at it again, the paragraph is pretty short so I have removed it.

File: 071-GitHubCopilot/Student/Challenge-05.md (hunk 8 above)
Comment (Whowong) [reply 071-GitHubCopilot/Student/Challenge-05.md]: I have removed the * in the end from the earlier comment.  The section is fairly small so it seems excessive.

**Review by jordanbean-msft — APPROVED** [review]

File: 071-GitHubCopilot/Coach/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+## Coach Prerequisites
+
+This hack has pre-reqs that a coach is responsible for understanding and/or setting up BEFORE hosting an event. Please review the [What The Hack Hosting Guide](https://aka.ms/wthhost) for information on how to host a hack event.
+
+The guide covers the common preparation steps a coach needs to do before any What The Hack event, including how to properly configure Microsoft Teams.
+
+### Student Resources
+
+Always refer students to the [What The Hack website](https://aka.ms/wth) for the student guide: [https://aka.ms/wth](https://aka.ms/wth)
```
[the remark this answers is not in the document]
Comment (Whowong) [reply 071-GitHubCopilot/Coach/README.md:32]: That seems to be from the template.  My hunch is one is just linking the text "What The Hack website" while the other is also giving the aka.ms link in the visible text.

File: 071-GitHubCopilot/Coach/Solution-00.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+
+**[Home](./README.md)** - [Next Solution >](./Solution-01.md)
+
+## Notes & Guidance
+
+A GitHub account, GitHub Copilot, and VS Code is all that is needed for this hack.  All are free and GitHub Copilot if needed has a free 30 day trial.
+
+- [Create GitHub Account](https://github.com/join)
+- [Install Visual Studio Code](../../000-HowToHack/WTH-Common-Prerequisites.md#visual-studio-code)
+- [Install GitHub Copilot in VS Code](https://docs.github.com/en/copilot)
```
[the remark this answers is not in the document]
Comment (Whowong) [reply 071-GitHubCopilot/Coach/Solution-00.md]: I like it!  I have updated

File: 071-GitHubCopilot/Coach/Solution-03.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+  - SQL Queries
+- To generate a commit message, they can go to VS Code, and in the source control section there will be these stars/diamonds next to the commit message box.  Selecting that will allow you to leverage Copilot to generate a commit message.  They can generate this after creating the pipeline to have something to checkin to.
+- This could also be a great time to talk about some of GitHub Copilot Enterprises other features such as PR message creation which is along the same concept. However this is a higher license tier and not part of the hack.  There are three tiers of GitHub Copilot Licensing, this hack will focus on the features that apply to Individual/Business:
+  - Copilot Individual ($10/Month) - This is for individual developers.
+  - Copilot Business ($19/Month) -  This is meant for organizations and allows capabilities to manage licenses and features within Copilot.
+  - Copilot Enterprise ($39/Month) - This requires the codebase to be in GitHub but allows for organizations to infuse AI across the developer workflow.
+  - Full Feature Comparison: https://github.com/features/copilot#pricing
+[Sample Pipeline Solution](./Solutions/Solution-03.yaml)
```
[the remark this answers is not in the document]
Comment (Whowong) [reply 071-GitHubCopilot/Coach/Solution-03.md:17]: That can be confusing or missed.  I have moved it to its own line.

**Review by kriation — APPROVED** [review]

**Pull request merged**